plugins/SimulationView/LayerSlider.qml

```diff
@@ -40,33 +40,41 @@ Item {
 
     property bool layersVisible: true
 
-    function getUpperValueFromSliderHandle () {
+    function getUpperValueFromSliderHandle() {
         return upperHandle.getValue()
     }
 
-    function setUpperValue (value) {
+    function setUpperValue(value) {
         upperHandle.setValue(value)
         updateRangeHandle()
     }
 
-    function getLowerValueFromSliderHandle () {
+    function getLowerValueFromSliderHandle() {
         return lowerHandle.getValue()
     }
 
-    function setLowerValue (value) {
+    function setLowerValue(value) {
         lowerHandle.setValue(value)
         updateRangeHandle()
     }
 
-    function updateRangeHandle () {
+    function updateRangeHandle() {
         rangeHandle.height = lowerHandle.y - (upperHandle.y + upperHandle.height)
     }
 
     // set the active handle to show only one label at a time
-    function setActiveHandle (handle) {
+    function setActiveHandle(handle) {
         activeHandle = handle
     }
 
+    function normalizeValue(value) {
+        if (value > sliderRoot.maximumValue)
+            return sliderRoot.maximumValue
+        else if (value < sliderRoot.minimumValue)
+            return sliderRoot.minimumValue
+        return value
```

Comment:
I would just use the JavaScript classic: `return Math.min(Math.max(value, upper), lower);`

Reply:
I avoid `else if`s whenever possible. I don't think they should be in any language. 😛

Reply:
Fair enough. But I would rather use `return Math.min(Math.max(value, lower), upper)` 😛

Done in 1a189ba4f5be0d0a377ddcdd8671e0a59b5d7533